docs/operator-manual/upgrading/2.14-3.0.md

```diff
@@ -209,6 +209,50 @@ Explanation:
 - In Helm 3.17.1, this behavior changed and `null` object in `values.yaml` still results in this warning upon performing `helm template`, but the resulting K8s object will be overridden with the invalid `null` value.
 - To resolve the issue, identify `values.yaml` with `null` object values, and remove those `null` values.   
 
+### Use Annotation-Based Tracking by Default
+
+The default behavior for [tracking resources](../../user-guide/resource_tracking.md) has changed to use annotation-based 
+tracking instead of label-based tracking. Annotation-based tracking is more reliable and less prone to errors caused by 
+external code copying tracking labels from one resource to another.
+
+#### Detection
+
+To detect if you are impacted, check the `argocd-cm` ConfigMap for the `application.resourceTrackingMethod` field. If it 
+unset or is set to `label`, you are using label-based tracking. If it is set to `annotation`, you are already using
+annotation-based tracking and are not impacted by this change.
+
+```sh
+kubectl get cm argocd-cm -n argocd -o jsonpath='{.data.application\.resourceTrackingMethod}'
+```
+
+#### Remediation
+
+For most users, it is safe to upgrade to Argo CD 3.0 and use annotation-based tracking. Labels will be replaced with
+annotations on the next sync. Applications will not be marked as out-of-sync if the labels are not present on the
+resources.
+
+Some users rely on label-based tracking to track resources that are not managed by Argo CD. They may set annotations
+to have Argo CD ignore the resource as extraneous or to disable pruning. If you are using label-based tracking to track
+resources that are not managed by Argo CD, you will need to construct tracking annotations instead of tracking labels
+and apply them to the relevant resources. The format of the tracking annotation is:
+
+```yaml
+argocd.argoproj.io/tracking-id: <app name>:<resource group>/<resource kind>:<resource namespace>/<resource name>
+```
+
+For cluster-scoped resources, the namespace is set to the value in the Application's `spec.destination.namespace` field.
```

Comment:
Isn't it set to `""` by your recent fix in https://github.com/argoproj/argo-cd/pull/22325/files#diff-10c7b3329ba1baf707f1917325609870201d61c16fb8dbcc787c421896bd4319R425-R426

```
For cluster-scoped resources, the namespace is set to the value in the Application's `spec.destination.namespace` field.
```

Reply:
@agaudreault the input to the tracking annotation construction code is set to `""`, but then that code replaces the empty string with the default namespace.

I don't like it, but I guess it doesn't really break anything. :-P